http-server/src/main/java/com/facebook/airlift/http/server/ClassPathResourceHandler.java

```diff
@@ -132,24 +130,54 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
             resourceStream = resource.openStream();
 
             String contentType = MIME_TYPES.getMimeByExtension(resource.toString());
-            response.setContentType(contentType);
-            extraHeaders.forEach((name, value) -> response.setHeader(name, value));
-
+            response.getHeaders().add(CONTENT_TYPE, contentType);
+            extraHeaders.forEach((name, value) -> response.getHeaders().add(name, value));
             if (skipContent) {
-                return;
+                callback.succeeded();
+                return true;
             }
 
-            ByteStreams.copy(resourceStream, response.getOutputStream());
+            try (OutputStream out = new BufferedOutputStream(new OutputStream()
```

Comment:
We could use 
```suggestion
OutputStream out = Response.asBufferedOutputStream(request, response)
```

instead of making our own buffered output stream